src/main/java/hudson/plugins/sonar/action/SonarCacheAction.java

```diff
@@ -105,6 +104,12 @@ static boolean isEntryValid(@Nullable ProjectInformation cached, long lastBuild)
     return false;
   }
 
+  @VisibleForTesting
+  protected void cacheProjectInfo(List<ProjectInformation> list, Long time) {
+    lastProjInfo = list;
+    lastRequest = time;
+  }
+
```

Comment:
I'm using this workaround so I don't have to mock `System.currentTimeMillis()` in the unit tests. Not sure if there's a better way, though...